base/src/main/java/io/spine/validate/MaxConstraint.java

```diff
@@ -32,11 +32,11 @@
         super(optionValue, maxRange(optionValue));
     }
 
-    private static <V extends Number & Comparable> Range<V> maxRange(MaxOption option) {
+    private static Range<StringDescribedNumber> maxRange(MaxOption option) {
         boolean inclusive = !option.getExclusive();
-        V minValue = fromOption(option.getValue());
+        StringDescribedNumber maxValue = new StringDescribedNumber(option.getValue());
```

Comment:
Please rename `StringDescribedNumber`. It's too long and starts with `String.` How about `NumberText`? Think of some alternatives, too.